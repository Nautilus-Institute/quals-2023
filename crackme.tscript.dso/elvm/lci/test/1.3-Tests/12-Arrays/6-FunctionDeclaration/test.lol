HAI 1.3
	I HAS A foo ITZ A BUKKIT
	foo HAS A var1 ITZ A BUKKIT
	foo HAS A var2 ITZ A BUKKIT
	foo HAS A var3 ITZ A BUKKIT

	HOW IZ foo fun1
		VISIBLE "a"
	IF U SAY SO

	HOW IZ foo fun2 YR a
		VISIBLE "b" AN a
	IF U SAY SO

	HOW IZ foo fun3 YR a AN YR b AN YR c
		VISIBLE "c" AN a AN b AN c
	IF U SAY SO

	HOW IZ foo'Z var1 fun1
		VISIBLE "i"
	IF U SAY SO

	HOW IZ foo'Z var1 fun2 YR a
		VISIBLE "j" AN a
	IF U SAY SO

	HOW IZ foo'Z var1 fun3 YR a AN YR b AN YR c
		VISIBLE "k" AN a AN b AN c
	IF U SAY SO

	I IZ foo'Z fun1 MKAY
	I IZ foo'Z fun2 YR 1 MKAY
	I IZ foo'Z fun3 YR 1 AN YR 2.345 AN YR "xyz" MKAY
	I IZ foo'Z var1'Z fun1 MKAY
	I IZ foo'Z var1'Z fun2 YR 1 MKAY
	I IZ foo'Z var1'Z fun3 YR 1 AN YR 2.345 AN YR "xyz" MKAY
KTHXBYE
