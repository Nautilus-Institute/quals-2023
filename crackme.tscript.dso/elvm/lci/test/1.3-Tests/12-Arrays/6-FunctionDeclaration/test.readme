This test checks that a function can be declared in an array slot.
