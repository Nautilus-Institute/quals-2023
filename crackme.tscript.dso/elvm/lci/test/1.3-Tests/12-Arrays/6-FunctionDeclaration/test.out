a
b1
c12.34xyz
i
j1
k12.34xyz
